src/go/rdctl/cmd/createProfile.go

```diff
@@ -0,0 +1,157 @@
+		}
+		result, err := createProfile()
+		if err != nil {
+			return err
+		}
+		fmt.Println(result)
+		return nil
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(createProfileCmd)
+	createProfileCmd.Flags().StringVar(&outputSettingsFlags.Format, "output", "", fmt.Sprintf("output format: %s|%s", plistFormat, regFormat))
+	createProfileCmd.Flags().StringVar(&outputSettingsFlags.RegistryHive, "hive", "", fmt.Sprintf(`registry hive: %s|%s (default "%s")`, reg.HklmRegistryHive, reg.HkcuRegistryHive, reg.HklmRegistryHive))
+	createProfileCmd.Flags().StringVar(&outputSettingsFlags.RegistryProfileType, "type", "", fmt.Sprintf(`registry section: %s|%s (default "%s")`, defaultsType, lockedType, defaultsType))
+	createProfileCmd.Flags().StringVar(&InputFile, "input", "", "File containing a JSON document (- for standard input)")
+	createProfileCmd.Flags().StringVarP(&JSONBody, "body", "b", "", "Command-line option containing a JSON document")
+	createProfileCmd.Flags().BoolVar(&UseCurrentSettings, "from-settings", false, "Use current settings")
+}
+
+func createProfile() (string, error) {
+	err := validateProfileFormatFlags()
+	if err != nil {
+		return "", err
+	}
+	var output []byte
+
+	if JSONBody != "" {
+		output = []byte(JSONBody)
+	} else if InputFile != "" {
+		if InputFile == "-" {
+			output, err = ioutil.ReadAll(os.Stdin)
+		} else {
+			output, err = ioutil.ReadFile(InputFile)
+		}
+	} else {
+		if !UseCurrentSettings {
+			// This should have been caught in validateProfileFormatFlags
+			return "", fmt.Errorf(`no output format specified: must specify exactly one output format of '--input FILE|-', '--body|-b STRING', or '--from-settings'`)
+		}
+		output, err = processRequestForUtility(doRequest("GET", versionCommand("", "settings")))
+	}
+	if err != nil {
+		return "", err
+	}
+	if outputSettingsFlags.Format == regFormat {
+		lines, err := reg.JsonToReg(outputSettingsFlags.RegistryHive, outputSettingsFlags.RegistryProfileType, string(output))
+		if err != nil {
+			return "", err
+		}
+		return strings.Join(lines, "\n"), nil
+	} else if outputSettingsFlags.Format == plistFormat {
+		return plist.JsonToPlist(string(output))
+	}
+	return "", fmt.Errorf(`internal error: expecting an output format of %s or %s, got %s`, regFormat, plistFormat, outputSettingsFlags.Format)
+}
+
+func validateProfileFormatFlags() error {
+	if outputSettingsFlags.Format == "" {
+		return fmt.Errorf(`an '--output FORMAT' option of either "%s" or "%s" must be specified`, plistFormat, regFormat)
+	}
+	if outputSettingsFlags.Format != plistFormat && outputSettingsFlags.Format != regFormat {
+		return fmt.Errorf(`received unrecognized '--output FORMAT' option of %s; "%s" or "%s" must be specified`, outputSettingsFlags.Format, plistFormat, regFormat)
+	}
+	if InputFile == "" && JSONBody == "" && !UseCurrentSettings {
+		return fmt.Errorf(`no output format specified: must specify exactly one output format of '--input FILE|-', '--body|-b STRING', or '--from-settings'`)
```

Comment:
This is not an "output" format, but an "input" source (maybe there is a better word than "source" for this?):

```suggestion
		return fmt.Errorf(`no input source specified: must specify exactly one input source from '--input FILE|-', '--body|-b STRING', or '--from-settings'`)
```

Same issue with the "too many" error message just below.